sylt-compiler/src/typechecker.rs

```diff
@@ -644,8 +646,11 @@ impl TypeChecker {
             }
 
             S::ExternalDefinition { var, ty, span, .. } => {
-                let ty = self.resolve_type(ctx, ty)?;
-                self.unify(*span, ctx, self.variables[*var].ty, ty)?;
+                self.external_definition(ctx, var, ty, span)?;
+            }
+
+            S::LuaDefinition { var, ty, span, .. } => {
+                self.external_definition(ctx, var, ty, span)?;
```

Comment:
Alternatively you could have combined the cases - it's fine this way though. But I'm not sure the function warrents its own existence.

Reply:
I made it a function because I wanted inline lua to work as inner statements, but it doesn't work there anyway...